editor/tree_search.cpp

```diff
@@ -68,10 +68,16 @@ void TreeSearch::_filter_tree(const String &p_search_mask) {
 		}
 
 		TreeItem *first_counting_ancestor = cur_item;
+		bool parents_visible = true;
 		while (first_counting_ancestor && !number_matches.has(first_counting_ancestor)) {
+			// Performance: we only need to check the first visible ancestor. This is already optimal because of the ordering in ordered_tree_items.
+			if (!first_counting_ancestor->is_visible()){
```

Comment:
This could sit happily in the while-condition. But I *feel* like the additional is_visible api-call does introduce minuscule, tho noticeable, amount of lag.

Reply:
I don't feel *any* lag filtering with this solution. Tho, if this does have a greater impact, we could also track seen items by explicitly marking their pointers in a local `HashMap`.

Reply:
I checked the godot source. TreeItem::is_visible simply returns visible. My *feeling* was probably unfounded. The api calls through GDExtension should not matter in the greater context. Still, the this solution not good.